app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java

```diff
@@ -685,77 +757,134 @@ private double getSumDateOffHaveSalary() {
-            mRequestDayOffTypesAttribute.setDestroy(FALSE);
-            mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+            setDestroyDayOffTypesAttribute();
         }
-        if (mRequestOff.getMiscarriageLeave() != null && !"".equals(
-                mRequestOff.getMiscarriageLeave())) {
-            mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.MISCARRIAGE_LEAVE);
+        if (specialDayOffSettingId.equals(TypeOfDays.MISCARRIAGE_LEAVE)) {
             mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getMiscarriageLeave());
-            mRequestDayOffTypesAttribute.setDestroy(FALSE);
-            mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+            setDestroyDayOffTypesAttribute();
         }
-        if (mRequestOff.getSickLeave() != null && !"".equals(mRequestOff.getSickLeave())) {
-            mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.SICK_LEAVE);
+        if (specialDayOffSettingId.equals(TypeOfDays.SICK_LEAVE)) {
             mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getSickLeave());
-            mRequestDayOffTypesAttribute.setDestroy(FALSE);
-            mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+            setDestroyDayOffTypesAttribute();
         }
-        if (mRequestOff.getPregnancyExaminationLeave() != null && !"".equals(
-                mRequestOff.getPregnancyExaminationLeave())) {
-            mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.PREGNANCY_EXAMINATON);
+        if (specialDayOffSettingId.equals(TypeOfDays.PREGNANCY_EXAMINATON)) {
             mRequestDayOffTypesAttribute.setNumberDayOff(
                     mRequestOff.getPregnancyExaminationLeave());
-            mRequestDayOffTypesAttribute.setDestroy(FALSE);
-            mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+            setDestroyDayOffTypesAttribute();
+        }
+        if (mRequestOff != null
+                && mRequestOff.getRequestDayOffTypes() != null
+                && mRequestOff.getRequestDayOffTypes().size() != 0) {
+            for (int i = 0; i < mRequestOff.getRequestDayOffTypes().size(); i++) {
+                if (mRequestOff.getRequestDayOffTypes().get(i).getSpecialDayOffSettingId()
+                        == Integer.parseInt(specialDayOffSettingId)) {
+                    mRequestDayOffTypesAttribute.setId(
+                            mRequestOff.getRequestDayOffTypes().get(i).getId());
+                }
+            }
+            if (specialDayOffSettingId.equals(TypeOfDays.LEAVE_FOR_MARRIAGE)) {
+                if (leaveForMarriageIsEmpty()) {
+                    mRequestDayOffTypesAttribute.setDestroy(TRUE);
+                }
+            }
+            if (specialDayOffSettingId.equals(TypeOfDays.LEAVE_FOR_CHILD_MARRIAGE)) {
+                if (leaveForChildMarriageIsEmpty()) {
```

Comment:
WARNING: [CollapsibleIfStatements] These nested if statements could be combined
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/basic.html#CollapsibleIfStatements